official/vision/beta/projects/yolo/modeling/backbones/Darknet.py

```diff
@@ -0,0 +1,356 @@
+        ["DarkRes", "residual", 8, False, 512, None, None, None, "leaky", -1, 4, True],  # 3
+        ["DarkRes", "residual", 4, False, 1024, None, None, None, "leaky", -1, 5, True],  # 6
+    ]
+}
+
+CSPDARKNETTINY = {
+    "list_names": LISTNAMES,
+    "splits": {"backbone_split": 28},
+    "backbone": [
+        ["DarkConv", None, 1, False, 32, 3, 2, "same", "leaky", -1, 0, False],  # 1
+        ["DarkConv", None, 1, False, 64, 3, 2, "same", "leaky", -1, 1, False],  # 1
+        ["CSPTiny", "csp_tiny", 1, False, 64, 3, 2, "same", "leaky", -1, 2, False],  # 3
+        ["CSPTiny", "csp_tiny", 1, False, 128, 3, 2, "same", "leaky", -1, 3, False],  # 3
+        ["CSPTiny", "csp_tiny", 1, False, 256, 3, 2, "same", "leaky", -1, 4, True],  # 3
+        ["DarkConv", None, 1, False, 512, 3, 1, "same", "leaky", -1, 5, True],  # 1
+    ]
+}
+
+DARKNETTINY = {
+    "list_names": LISTNAMES,
+    "splits": {"backbone_split": 14},
+    "backbone": [
+        ["DarkConv", None, 1, False, 16, 3, 1, "same", "leaky", -1, 0, False],  # 1
+        ["DarkTiny", None, 1, True, 32, 3, 2, "same", "leaky", -1, 1, False],  # 3
+        ["DarkTiny", None, 1, True, 64, 3, 2, "same", "leaky", -1, 2, False],  # 3
+        ["DarkTiny", None, 1, False, 128, 3, 2, "same", "leaky", -1, 3, False],  # 2
+        ["DarkTiny", None, 1, False, 256, 3, 2, "same", "leaky", -1, 4, True],
+        ["DarkTiny", None, 1, False, 512, 3, 2, "same", "leaky", -1, 5, False],  # 3
+        ["DarkTiny", None, 1, False, 1024, 3, 1, "same", "leaky", -1, 5, True],  # 6  #route
+    ]
+}
+
+BACKBONES = {
+    "darknettiny": DARKNETTINY,
+    "darknet53": DARKNET53,
+    "cspdarknet53": CSPDARKNET53,
+    "cspdarknettiny": CSPDARKNETTINY
+}
+
+
+@ks.utils.register_keras_serializable(package='yolo')
+class Darknet(ks.Model):
+
+  def __init__(
+      self,
+      model_id="darknet53",
+      input_shape=tf.keras.layers.InputSpec(shape=[None, None, None, 3]),
+      min_size=None,
+      max_size=5,
+      activation=None,
+      use_sync_bn=False,
+      norm_momentum=0.99,
+      norm_epsilon=0.001,
+      kernel_initializer='glorot_uniform',
+      kernel_regularizer=None,
+      bias_regularizer=None,
+      config=None,
+      **kwargs):
+
+    layer_specs, splits = Darknet.get_model_config(model_id)
+
+    self._model_name = model_id
+    self._splits = splits
+    self._input_shape = input_shape
+    self._registry = layer_registry()
+
+    # default layer look up
+    self._min_size = min_size
+    self._max_size = max_size
+    self._output_specs = None
+
+    self._kernel_initializer = kernel_initializer
+    self._bias_regularizer = bias_regularizer
+    self._norm_momentum = norm_momentum
+    self._norm_epislon = norm_epsilon
+    self._use_sync_bn = use_sync_bn
+    self._activation = activation
+    self._weight_decay = kernel_regularizer
+
+    self._default_dict = {
+        "kernel_initializer": self._kernel_initializer,
+        "weight_decay": self._weight_decay,
```

Comment:
Can you keep the naming kernel_regularizer?

Reply:
addressed